src/panels/lovelace/cards/hui-sensor-card.ts

```diff
@@ -55,43 +58,65 @@ class HuiSensorCard extends EventsMixin(LitElement) {
     this._config = cardConfig;
   }
 
-  shouldUpdate(changedProps) {
-    const change = changedProps.has("_entity") || changedProps.has("_line");
-    return change;
+  public getCardSize(): number {
+    return 3;
   }
 
-  render({ _entity, _line } = this) {
+  protected render(): TemplateResult {
+    if (!this._config || !this.hass) {
+      return html``;
+    }
+
+    const stateObj = this.hass.states[this._config.entity];
+    if (!stateObj) {
+      return html``;
+    }
+
     return html`
-      ${this._style()}
+      ${this.renderStyle()}
       <ha-card @click="${this._handleClick}">
         <div class="flex">
           <div class="icon">
-            <ha-icon .icon="${this._computeIcon(_entity)}"></ha-icon>
+            <ha-icon
+              .icon="${this._config.icon || stateIcon(stateObj)}"
+            ></ha-icon>
           </div>
           <div class="header">
-            <span class="name">${this._computeName(_entity)}</span>
+            <span class="name"
+              >${this._config.name || computeStateName(stateObj)}</span
+            >
           </div>
         </div>
         <div class="flex info">
-          <span id="value">${_entity.state}</span>
-          <span id="measurement">${this._computeUom(_entity)}</span>
+          <span id="value">${stateObj.state}</span>
+          <span id="measurement"
+            >${
+              this._config.unit || stateObj.attributes.unit_of_measurement
+            }</span
+          >
         </div>
         <div class="graph">
           <div>
             ${
-              _line
-                ? svg`
-                    <svg width="100%" height="100%" viewBox="0 0 500 100">
-                      <path
-                        d="${_line}"
-                        fill="none"
-                        stroke="var(--accent-color)"
-                        stroke-width="5"
-                        stroke-linecap="round"
-                        stroke-linejoin="round"
-                      />
-                    </svg>
-                  `
+              this._config.graph !== "none" &&
+              stateObj.attributes.unit_of_measurement
+                ? until(
+                    this._getHistory().then((history) => {
+                      return svg`
+                        <svg width="100%" height="100%" viewBox="0 0 500 100">
+                          <path
+                            d="${history}"
+                            fill="none"
+                            stroke="var(--accent-color)"
+                            stroke-width="5"
+                            stroke-linecap="round"
+                            stroke-linejoin="round"
+                          />
+                        </svg>
+                      `;
+                    }),
+                    html``
```

Comment:
An error for no `unit_of_measurement` could save users time in debugging here as well. My thinking is, the more clear errors we can show, the less questions in chat and issues in git :)

Reply:
If we were to do that then we would want to not default the graph. Meaning only show the graph if `config.graph === "line"` 

Dont want an error by default

Reply:
![image](https://user-images.githubusercontent.com/18730868/49236060-5849ae80-f3c9-11e8-8327-43ffd183a40b.png)

I think by default it should not show the graph. Graph should be an opt in so I agree with you @iantrich 

Now will @balloob agree